operator/k8s/resource_ctors.go

```
@@ -57,3 +57,16 @@ func CiliumEndpointSliceResource(lc hive.Lifecycle, cs client.Clientset, opts ..
 	)
 	return resource.New[*cilium_api_v2alpha1.CiliumEndpointSlice](lc, lw, resource.WithMetric("CiliumEndpointSlice")), nil
 }
+
+func CiliumNodeResource(lc hive.Lifecycle, cs client.Clientset, opts ...func(*metav1.ListOptions)) (resource.Resource[*cilium_api_v2.CiliumNode], error) {
```

nit: do we need to return an error here, the `cell.Provide(ctors...)` can take either?

---

Currently, there is no case where the ctor can fail, but if an error occurs, the hive calls log.Fatal with a message like:

`failed to build resource.Resource[*github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2.CiliumNode]: received non-nil error from function \"github.com/cilium/cilium/operator/k8s\".CiliumNodeResource`